uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/layer/impl/DefaultGridLayerTest.java

```diff
@@ -261,4 +270,60 @@ private void checkConnectorsVisibility(final boolean isVisible) {
         assertEquals(isVisible,
                      connectors.iterator().next().isVisible());
     }
+
+    @Test
+    public void testRegister() {
+        final GridWidget gridWidget1 = mock(GridWidget.class);
+        final GridWidget gridWidget2 = mock(GridWidget.class);
+
+        gridLayer.register(gridWidget1);
+
+        assertThat(gridLayer.getGridWidgets().size()).isEqualTo(1);
+        assertThat(gridLayer.getGridWidgets()).contains(gridWidget1);
+
+        gridLayer.register(gridWidget2);
+
+        assertThat(gridLayer.getGridWidgets().size()).isEqualTo(2);
+        assertThat(gridLayer.getGridWidgets()).contains(gridWidget1, gridWidget2);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testRegisterAndAddAsPrimitive() {
+        final GridWidget gridWidget1 = mock(GridWidget.class);
+        final GridWidget gridWidget2 = mock(GridWidget.class);
+        when(gridWidget1.asNode()).thenReturn(mock(Node.class));
+        when(gridWidget1.getModel()).thenReturn(new BaseGridData());
+
+        gridLayer.add(gridWidget1);
+
+        assertThat(gridLayer.getGridWidgets().size()).isEqualTo(1);
+        assertThat(gridLayer.getGridWidgets()).contains(gridWidget1);
+
+        gridLayer.register(gridWidget2);
```

Comment:
Wasn't your intention to use `gridLayer.add(grdiWidget2)` here?

Reply:
No. I "add" one ```GridWidget``` and only "register" the other. The two operations should result in there being two grids returned from ```getGridWidgets()``` (the "add" is the outer most grid added to the layer, and "registrations" are nested grids that are added to the layer by the Column Renderer - in DMN).